galley/pkg/config/analysis/analyzers/util/find_errorline_utils.go

```diff
@@ -0,0 +1,110 @@
+	// Use the path template to find the exact line number for the field
+
+	// Path for host in VirtualService.
+	// Required parameters: route rule, route rule index, route index.
+	DestinationHost = "{.spec.%s[%d].route[%d].destination.host}"
+
+	// Path for mirror host in VirtualService.
+	// Required parameters: http index.
+	MirrorHost = "{.spec.http[%d].mirror.host}"
+
+	// Path for VirtualService gateway.
+	// Required parameters: gateway index.
+	VSGateway = "{.spec.gateways[%d]}"
+
+	// Path for regex match of uri, scheme, method and authority.
+	// Required parameters: http index, match index, where to match.
+	URISchemeMethodAuthorityRegexMatch = "{.spec.http[%d].match[%d].%s.regex}"
+
+	// Path for regex match of headers and queryParams.
+	// Required parameters: http index, match index, where to match, match key.
+	HeaderAndQueryParamsRegexMatch = "{.spec.http[%d].match[%d].%s.%s.regex}"
+
+	// Path for regex match of allowOrigins.
+	// Required parameters: http index, allowOrigins index.
+	AllowOriginsRegexMatch = "{.spec.http[%d].corsPolicy.allowOrigins[%d].regex}"
+
+	// Path for workload selector.
+	// Required parameters: selector label.
+	WorkloadSelector = "{.spec.workloadSelector.labels.%s}"
+
+	// Path for port from ports collections.
+	// Required parameters: port index.
+	PortInPorts = "{.spec.ports[%d].port}"
+
+	// Path for fromRegistry in the mesh networks.
+	// Required parameters: network name, endPoint index.
+	FromRegistry = "{.networks.%s.endpoints[%d]}"
+
+	// Path for the image in the container.
+	// Required parameters: container index.
+	ImageInContainer = "{.spec.containers[%d].image}"
+
+	// Path for namespace in metadata.
+	// Required parameters: none.
+	MetadataNamespace = "{.metadata.namespace}"
+
+	// Path for name in metadata.
+	// Required parameters: none.
+	MetadataName = "{.metadata.name}"
+
+	// Path for namespace in authorizationPolicy.
+	// Required parameters: rule index, from index, namespace index.
+	AuthorizationPolicyNameSpace = "{.spec.rules[%d].from[%d].source.namespaces[%d]}"
+
+	// Path for annotation.
+	// Required parameters: annotation name.
+	Annotation = "{.metadata.annotations.%s}"
+
+	// Path for selector in Gateway.
+	// Required parameters: selector label.
+	GatewaySelector = "{.spec.selector.%s}"
+
+	// Path for credentialName.
+	// Required parameters: server index.
+	CredentialName = "{.spec.servers[%d].tls.credentialName}"
+)
+
+// ErrorLine returns the line number of the input path key in the resource
+func ErrorLine(r *resource.Instance, path string) (line int, found bool) {
+	return findErrorLine(path, r.Origin.FieldMap())
+}
+
+// FindErrorLine returns the line number of the input key from the input map
+func findErrorLine(key string, m map[string]int) (line int, found bool) {
+	line, ok := m[key]
+	if !ok {
+		return 0, false
+	}
+	return line, true
+}
```

Comment:
I think you can combine these, findErrorLine doesn't really seem necessary anymore.